tests/test_sensor.py

```diff
@@ -1292,6 +1320,26 @@ async def test_last_feeding_size_sensor_v2(zha_gateway: Gateway) -> None:
     assert_state(entity, 5.0, "g")
 
 
+async def test_state_class(zha_gateway: Gateway) -> None:
+    """Test quirks defined sensor."""
+
+    zha_device, cluster = await zigpy_device_aqara_sensor_v2_mock(zha_gateway)
+    assert isinstance(zha_device.device, CustomDeviceV2)
+    power_entity = get_entity(zha_device, platform=Platform.SENSOR, qualifier="power")
+    energy_entity = get_entity(zha_device, platform=Platform.SENSOR, qualifier="energy")
+    energy_delivered_entity = get_entity(
+        zha_device, platform=Platform.SENSOR, qualifier="energy_delivered"
+    )
+    energy_invalid_state_class = get_entity(
+        zha_device, platform=Platform.SENSOR, qualifier="energy_invalid_state_class"
+    )
+
+    assert power_entity.state_class == SensorStateClass.MEASUREMENT
+    assert energy_entity.state_class == SensorStateClass.TOTAL
+    assert energy_delivered_entity.state_class == SensorStateClass.TOTAL_INCREASING
+    assert energy_invalid_state_class.state_class is None
```

Comment:
For device class validation, we test this inside [`test_discovery.py`](https://github.com/zigpy/zha/blob/2d1501309bc98ee5598953e4c0d9d0dd838da344/tests/test_discover.py#L665-L705) and check for the error text.
It's probably fine to leave this in the sensor tests here though. We could add an additional check for asserting that the "Quirks provided an invalid state class [...]" message is logged.

Reply:
I've added the log check.
Since this is for sensors only, I think it fits `test_sensor.py` better, but I am ok with moving it to `test_discovery.py` if you prefer.